File: matlab/ex_4.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%   @ Authors : SEKHRI Aymen
%               MOHAMMED HACENE Tarek
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

close all 
clear all 
clc 

image_name = 'rose.jpeg'

% load image
rose_img = imread(image_name)

% resize image using "nearest methode" interpolation 
resized_img = imresize(rose_img,[512,512],'nearest');

% plot original image
figure(1) 
imshow(rose_img);
title('original image')

% plot resized image
figure(2) 
imshow(resized_img);
title('resized + interpolated image with "nearest methode"') 

File: python/ex_4.py
####################################################  
#
# 	@ Authors : SEKHRI Aymen
# 				MOHAMMED HACENE Tarek
#	
# 	@ Hint: you have to install all requirements
# 			from requirements.txt                    
#
#################################################### 

import numpy as np
import cv2 as cv
import matplotlib.pyplot as plt


# load the image
onion_img = cv.imread("onion.png")

# Resized image based on Nearest method interpolation
resized_img = cv.resize(onion_img, (512, 512), interpolation = cv.INTER_NEAREST)

cv.imshow("Original Image", onion_img)
cv.imshow('Resized Image', resized_img)
cv.waitKey()
cv.destroyAllWindows()